base/titer_model.py

```diff
@@ -149,6 +150,25 @@ def filter_strains(titers, strains):
         return {key: value for key, value in titers.iteritems()
                 if key[0] in strains and key[1][0] in strains}
 
+    @classmethod
+    def subset_to_date(cls, titers, node_lookup, date_range):
+        """Subset the given titers to the given date range based on the dates annotated
+        on each node.
+        """
+        # Filter titers from the future by keeping titers associated with
+        # strains from the given timepoint or earlier.
+        filtered_strains = set([
+            node_name
+            for node_name, node in node_lookup.items()
+            if node.numdate <= date_range[1] and node.numdate >= date_range[0]
```

Comment:
doesn't matter much since these are floats, but `a<=x<b` would be more consistent rather than `>=` and `<=`